server/image_pull.go

```diff
@@ -54,8 +55,13 @@ func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (resp
 			}
 		}
 		options := &copy.Options{
-			SourceCtx: &types.SystemContext{},
+			SourceCtx: &types.SystemContext{
+				DockerRegistryUserAgent: useragent.Get(ctx),
+			},
 		}
+
+		logrus.Debugf("runcom %v", useragent.Get(ctx))
```

Comment:
Why are you calling useragent.Get twice.
Why not options.SourceCtx.DockerRegisteryUserAgent

Reply:
Oh sorry, this was just debug, removing